packages/react-native-reanimated/src/createAnimatedComponent/PropsFilter.tsx

```diff
@@ -67,16 +67,32 @@ export class PropsFilter implements IPropsFilter {
         has('workletEventHandler', value) &&
         value.workletEventHandler instanceof WorkletEventHandler
       ) {
-        if (value.workletEventHandler.eventNames.length > 0) {
-          value.workletEventHandler.eventNames.forEach((eventName) => {
-            props[eventName] = has('listeners', value.workletEventHandler)
-              ? (
-                  value.workletEventHandler.listeners as Record<string, unknown>
-                )[eventName]
-              : dummyListener;
-          });
+        const handler = value.workletEventHandler;
+        const isWebHandler = has('listeners', handler);
+        const hasJSHandlers = Object.keys(handler.JSHandlers).length > 0;
+
+        if (hasJSHandlers) {
+          if (isWebHandler) {
+            // on web, our and JS handlers are merged in listeners object
+            Object.keys(handler.listeners).forEach((eventName) => {
+              props[eventName] = handler.listeners[eventName];
+            });
+          } else {
+            // on mobile platforms, we just set the JS handlers to the props
+            Object.keys(handler.JSHandlers).forEach((eventName) => {
+              props[eventName] = handler.JSHandlers[eventName];
+            });
+          }
         } else {
-          props[key] = dummyListener;
+          if (handler.eventNames.length > 0) {
```

Comment:
I'm not sure about this part of the code. It seems like if there is any JSHandler, we ignore all worklet event handlers.

Reply:
No we don't! They don't need to be attached anywhere. The only logic concerning worklet event handlers was putting `dummyListeners` in places for their respected props. Worklet event handlers are registered in native modules and don't need props to work.

Reply:
And if we have JS handlers, we can't always just put `dummyListeners` in the props because some JS handlers may need to use that. This is why there are 2 different approaches depending on JS handlers presence

Reply:
But I agree, I might've forgot to add `dummyListeners` for only worklet-event-handler events